### ISSUE TITLE
Shapefile Datasource: after edit shapefile, how to make it work without rerun the entire app?

### ISSUE THREAD

**Issue by ccFiona** [mapnik/mapnik#3819]
Scenairo: 
mapnik 3.0.8 , using shape as datasource (for it's faster then using postgis as datasource)

now i regularly need to edit  the content of the shapefiles, while keeping shapefiles' file name(ie: uri) and the config xml unchanged.  But what mapnik gives is still the old pics, the edition doesn't take effect, unless i rerun my entire application.  

I wonder if there is any interface I can use to reset mapnik's shapefile datasourse content?  Or what should I do to achieve my purpose? Thanks a lot!

**Comment by artemp** [comment]
can you provide a bit more context, pls. Are you using python bindings or ?

**Comment by ccFiona** [comment]
I'm using python, and I found **mapnik.clear_cache()** method to clear file cache.

**Comment by ccFiona** [comment]
I'm using nginx+uwsgi+flask+python+mapnik to provide tile service. **mapnik reads xml which uses shapefile as datasource** to render tiles.

> <Map srs="1">
>      <Style name="1*">
>      </Style>
>      <Layer name="**" srs="**">
>          <StyleName>***</StyleName>
>          <Datasource>
>              <Parameter name=**"file">/test/test.shp**</Parameter>
>              <Parameter name=**"type">shape**</Parameter>
>          </Datasource>
>      </Layer>
> </Map>
> 

My problem is that when shapefile content changes( e.g. shapefile name stays the same, delete an entry from the shapefile), **mapnik still uses the old  datasource even if I called mapnik.clear_cache() before rendering**.  
This is my **production environment: uwsgi-2.0.17.1, python-2.7.5, mapnik-3.0.21**
What's more weird, when I tested this problem only use python+mapnik, after changing shapefile and called mapnik.clear_cache() the tile result is all right.
Any thoughts on how to solve this problem is highly appreciated.
